### PR TITLE
asLib check against PeerInfo

### PR DESCRIPTION
**Pull request by mdavidsaver**
Check Put permissions against new PeerInfo API peer IP address and user or role/group names.

host and user are handled as with CA in HAG and UAG entires.  role/group may appear as special UAG entries prefixed with "role/".  eg.

```
UAG(special) { root, "role/admin", "role/wheel" }
```

Always compares against peer numeric IP address.  Matching against host names in .acf file requires https://code.launchpad.net/~epics-core/epics-base/+git/asLib/+merge/358822

### CODE REVIEW

**Comment by mdavidsaver** [comment]
cf. https://github.com/epics-base/pvAccessCPP/pull/136

**Comment by mdavidsaver** [comment]
@bhill-slac FYI

File: pdbApp/pdbgroup.cpp [hunk 1]
```diff
@@ -350,8 +360,10 @@ void PDBGroupPut::put(pvd::PVStructure::shared_pointer const & value,
         putpvif[i].reset(info.builder->attach(info.chan, value, info.attachment));
     }
 
-    pvd::Status ret;
-    if(atomic) {
+    if(!ret.isOK()) {
```
Comment (bhill-slac) [inline pdbApp/pdbgroup.cpp]: Why check reg.isOK here?   Nothing has touched ret since it's default constructor set it OK.

Comment (mdavidsaver) [reply pdbApp/pdbgroup.cpp]: I think this is a leftover for an earlier iteration where more happened in the loop above.  I'll remove it.

**Comment by bhill-slac** [comment]
Looks good to me.
Tested w/ IOC built using asLib epics-base pull request and pvAccess gwdev branch from mdavidsaver.

**Comment by mdavidsaver** [comment]
Thank you Bruce

**Pull request merged**